src/error.rs

```diff
@@ -174,3 +179,23 @@ impl AsPearlError for anyhow::Error {
         self.downcast_ref()
     }
 }
+
+/// Util function for UnexpectedEOF error conversion
+pub fn unexpected_eof_converter(e: AHError) -> AHError {
+    if let Some(io_error) = e.downcast_ref::<std::io::Error>() {
+        if let std::io::ErrorKind::UnexpectedEof = io_error.kind() {
+            return AHError::from(Error::bincode("unexpected eof".into()));
+        }
+    }
+    e
+}
+
+/// Util function for UnexpectedEOF error conversion with additional context
+pub fn unexpected_eof_converter_ctx(e: AHError, context: String) -> AHError {
```

Comment:
I think, this function is not needed. It is better to use `with_context` to set the context after the conversion of error happened